EXT_BeamPhysics.md

```diff
@@ -0,0 +1,346 @@
+
+- **Polar coordinates**: **Polar** coordinates are **(r, theta, phi)** where **r** is the radius, **theta** is the angle from the **z** or **s** axis, and **phi** is the projected azimuthal angle in the **(x, y)** plane.
+
+- **Particle Group**: The **Particle Group** is a group for specifying a set of particles such as the particles in a bunch. The Beam Physics extension defines a standard for the  **Particle Group** as discussed below.
+
+Notes:
+------
+
+- When using the **lattice** coordinate system, the `position` coordinates are **(x, y, s)** or **(x, y, z)** where, nominally, **x** is the "horizontal" component, **y** is the "vertical" coordinate, and **s** or **z** is the lattice longitudinal coordinate.
+
+Additional File Root Group (Path `/`) Attributes
+------------------------------------------------
+
+The following records are defined for the file root group.
+
+- `fileType`
+  - Type: Optional *(string)*
+  - Description: The type of data being stored in the file. If present, must be set to `openPMD`. This attribute is used in systems where different data files can contain different types of data and allows for quick identification of the what type of data is in a given file.
+
+- `latticeName`
+  - Type: Optional *(string)*
+  - Description: The name of the lattice.
+
+- `latticeFile`
+  - Type: Optional *(string)*
+  - Description: The location of the root lattice file.
+
+Particle Group Standard
+============================
+
+The **Particle Group** is a group for specifying a set of particles such as the particles in a bunch. Multiple **Particle Groups** can be defined in a file. The path for a **Particle Group** is given by the **basePath** and **particlesPath** attributes in the file root group as discussed in the base OpenPMD standard. For example, if **basePath** is  "/data/%T/", and **ParticlesPath** is "particles/", then **Particle Groups** paths would be of the form "/data/%T/particles/" where "%T" is an integer. EG: "/data/37/particles/".
+
+
+`Particle Group` Attributes
+--------------------------------
+
+For each **particle group** the following attributes are defined:
+
+- `chargeLive`
+  - Type: Optional *(real)*
+  - Description: The total charge of all the live particles.
+
+- `chargeUnitSI`
+  - Type: *(real)*. Only required if `chargeLive` or `totalCharge` is present.
+  - Description: Unit conversion factor to multiply `chargeLive` or `totalCharge` by in order to convert to SI units.
+
+- `latticeElementName`
+  - Type: Optional *(string)*
+  - Description: The name of the lattice element the particle are in. This only makes sense if all particles are in the same element. [Keep in mind that if particles are lost and the lost particles are also included in the file, not all particles may be in the same element.] Also see: `locationInElement`.
+
+- `numParticles`
```

Comment:
Is this strictly needed or can this be inferred by just querying any record in the particle group and its file-format provided extent?

Reply:
Unfortunately it cannot always be inferred.